contracts/debt/IncurDebtV1.sol

```diff
@@ -0,0 +1,453 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity ^0.8.10;
+
+import "../interfaces/IOHM.sol";
+import "../interfaces/IsOHM.sol";
+import "../interfaces/IgOHM.sol";
+import "../interfaces/IERC20.sol";
+import "../interfaces/IStaking.sol";
+import "./interfaces/ITreasury.sol";
+import "../libraries/SafeERC20.sol";
+import "../types/OlympusAccessControlledV2.sol";
+
+error IncurDebtV1_NotBorrower(address _borrower);
+error IncurDebtV1_InvaildNumber(uint256 _amount);
+error IncurDebtV1_WrongTokenAddress(address _token);
+error IncurDebtV1_AlreadyBorrower(address _borrower);
+error IncurDebtV1_AboveGlobalDebtLimit(uint256 _limit);
+error IncurDebtV1_AboveBorrowersDebtLimit(uint256 _limit);
+error IncurDebtV1_LimitBelowOutstandingDebt(uint256 _limit);
+error IncurDebtV1_AmountAboveBorrowerBalance(uint256 _amount);
+error IncurDebtV1_AmountMoreThanBorrowersLimit(uint256 _borrower);
+error IncurDebtV1_OHMAmountMoreThanAvailableLoan(uint256 _amount);
+error IncurDebtV1_BorrowerHasNoOutstandingDebt(address _borrower);
+error IncurDebtV1_BorrowerStillHasOutstandingDebt(address _borrower);
+
+contract IncurDebtV1 is OlympusAccessControlledV2 {
+    using SafeERC20 for IERC20;
+
+    event GlobalLimit(uint256 indexed _limit);
+    event BorrowerAllowed(address indexed _borrower);
+    event BorrowerRevoked(address indexed _borrower);
+    event BorrowerDebtLimitSet(address indexed _borrower, uint256 indexed _limit);
+    event CollateralInSOHMIncreased(address indexed _borrower, uint256 indexed _currentCollateralInSOHM);
+    event BorrowerDeposit(
```

Comment:
can't CollateralInSOHMIncreased and BorrowerDeposit be the same event? Both track depositing collateral and if we track deposit inflows we want to track them both as basically the same thing. Is having an event for the first time they deposited collateral that useful we already have events for when we allow them.

Reply:
I think it's important to track an initial deposit and also when a users collateral has been increased due to rebase. For future reference a need for these different information might come into play.